contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/HDF5BatchReader.java

```diff
@@ -0,0 +1,1164 @@
+                break;
+              case "long":
+                mapBuilder.add(info.getName(), TypeProtos.MinorType.BIGINT, TypeProtos.DataMode.REPEATED);
+                break;
+              case "String":
+                mapBuilder.add(info.getName(), TypeProtos.MinorType.VARCHAR, TypeProtos.DataMode.REPEATED);
+                break;
+            }
+          }
+          TupleMetadata finalInnerSchema = mapBuilder.resumeSchema().buildSchema();
+
+          index = rowWriter.tupleSchema().index(COMPOUND_DATA_FIELD_NAME);
+          if (index == -1) {
+            index = rowWriter.addColumn(finalInnerSchema.column(COMPOUND_DATA_FIELD_NAME));
+          }
+        }
+        TupleWriter listWriter = rowWriter.column(index).tuple();
+
+        //Note: The [][] returns an array of [rows][cols]
+        for (int row = 0; row < values.length; row++) {
+          // Iterate over rows
+          for (int col = 0; col < values[row].length; col++) {
+            assert fieldNames != null;
+            currentFieldName = fieldNames.get(col);
+            ArrayWriter innerWriter = listWriter.array(currentFieldName);
+            if (values[row][col] instanceof Integer) {
+              innerWriter.scalar().setInt((Integer) values[row][col]);
+            } else if (values[row][col] instanceof Short) {
+              innerWriter.scalar().setInt((Short) values[row][col]);
+            } else if (values[row][col] instanceof Byte) {
+              innerWriter.scalar().setInt((Byte) values[row][col]);
+            } else if (values[row][col] instanceof Long) {
+              innerWriter.scalar().setLong((Long) values[row][col]);
+            } else if (values[row][col] instanceof Float) {
+              innerWriter.scalar().setDouble((Float) values[row][col]);
+            } else if (values[row][col] instanceof Double) {
+              innerWriter.scalar().setDouble((Double) values[row][col]);
+            } else if (values[row][col] instanceof BitSet || values[row][col] instanceof Boolean) {
+              innerWriter.scalar().setBoolean((Boolean) values[row][col]);
+            } else if (values[row][col] instanceof String) {
+              innerWriter.scalar().setString((String) values[row][col]);
+            }
+            if (col == values[row].length) {
+              innerWriter.save();
+            }
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * First tries to get drill temporary directory value from from config ${drill.tmp-dir},
+   * then checks environmental variable $DRILL_TMP_DIR.
+   * If value is still missing, generates directory using {@link Files#createTempDir()}.
+   * If temporary directory was generated, sets {@link #deleteTmpDir} to true
+   * to delete directory on drillbit exit.
+   *
+   * @return drill temporary directory path
+   */
+  private File getTmpDir() {
+
+    DrillConfig config = DrillConfig.create();
```

Comment:
You should not create new DrillConfig, use config for Drillbit.

Reply:
@arina-ielchiieva I figured this out by moving this function to the actual formatPlugin class, you have access to the Drillbit.